integration-tests/bot/jail_test.py

```diff
@@ -0,0 +1,186 @@
+        print("check validators")
+        data =len(x)
+        print("count={}  check_validators={}".format(data,x))
+        return data
+      
+
+    def wait_for_ready(self,count) :
+        initial_time=time.time() # in seconds
+        MAX_TIME = 3600
+        while True:
+            current_time= time.time()
+            elasped_time= current_time - initial_time
+            remain_time = MAX_TIME - elasped_time
+            validators=self.check_validators()
+            if remain_time< 0 :
+                assert False
+            print("{0}  remain time={1:.2f}  current validators={2}  waiting for validators={3}".format(datetime.datetime.now(), remain_time, validators, count))
+            if count== validators :
+                print("validators ready")
+                break
+            time.sleep(10)
+
+
+    def test_jailing(self) :
+        print("test jailing")
+        self.wait_for_ready(2)
+        containers=self.get_containers()
+        print(containers)
+        if "jail_chain1_1" in containers :
+            assert True
+        else :
+            assert False
+        print("wait for jailing")
+        time.sleep(10)
+        jailthis = containers["jail_chain1_1"]
+        print("jail = " , jailthis)
+        jailthis.kill()
+        self.wait_for_ready(1)
+        #jailed
+        containers=self.get_containers()
+        print(containers)
+        if "jail_chain1_1" in containers :
+            assert False
+        else :
+            assert True 
+        print("jail test success")
+
+
+    def test_unjailing(self) :
+        initial_time=time.time() # in seconds
+        print("test unjailing")
+        self.wait_for_ready(1)
+
+        count=2
+        MAX_TIME = 3600  
+        while True:
+            current_time= time.time()
+            elasped_time= current_time - initial_time
+            remain_time = MAX_TIME - elasped_time
+            validators=self.check_validators()
+            if remain_time< 0 :
+                assert False
+            self.unjail("b","1", self.node1_address)
+            state= self.get_staking_state("b","1", self.node1_address)
+            print("state {}".format(state))
+            punishment=state["punishment"] 
+            print("{0}  remain time={1:.2f}  punishment {2}".format(datetime.datetime.now(), remain_time, punishment))
+            if punishment== None :
+                print("unjailed!!")
+                break
+            else :
+                print("still jailed")
+            time.sleep(10)
+        print("unjail test success")
+
+    ############################################################################3
+    def main (self) :
+        self.test_jailing()
+        try :
+            self.restore_wallets()
+        except:
+            print("wallet already exists")
+        self.create_addresses()
+        self.test_unjailing()
+
+
+    def read_info(self):
+        print("read data")
+        with open('nodes_info.json') as json_file:
```

Comment:
I think we don't need to write this file, we can recover staking addresses from mnemonic through rpc.

Reply:
it's for give information to other file.